### PR TITLE
Optimized with jdk new features

### PR DESCRIPTION
**Pull request by Phixsura**
Use new features of jdk17 to optimize code in 2022.x branch. 
#2971

### CODE REVIEW

File: spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/main/java/com/alibaba/cloud/sentinel/custom/SentinelBeanPostProcessor.java [hunk 1]
```diff
@@ -174,7 +173,7 @@ private boolean checkSentinelProtect(RootBeanDefinition beanDefinition,
 	private boolean checkMethodMetadataReadingVisitor(RootBeanDefinition beanDefinition) {
 		return beanDefinition.getSource() instanceof MethodMetadata
 				&& ((MethodMetadata) beanDefinition.getSource())
-						.isAnnotated(SentinelRestTemplate.class.getName());
```
Comment (123liuziming) [inline spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-sentinel/src/main/java/com/alibaba/cloud/sentinel/custom/SentinelBeanPostProcessor.java:177]: You may run `mvn clean install` to execute the checkstyle plugin before you submit a pull request. It seems that you didn't execute the command?

**Review by 123liuziming — COMMENTED** [review]
You may run `mvn clean install` to execute the checkstyle plugin before you submit a pull request. It seems that you didn't execute the command?

**Comment by Phixsura** [comment]
> You may run to execute the checkstyle plugin before you submit a pull request. It seems that you didn't execute the command?`mvn clean install`

I ran `mvn clean install` and it was no problem

**Review by 123liuziming — APPROVED** [review]
LGTM

**Review by steverao — APPROVED** [review]
LGTM, thanks @MieAh

**Pull request merged**